pkg/clients/edgeconnect/environment_settings.go

```diff
@@ -0,0 +1,156 @@
+package edgeconnect
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/clients/utils"
+	"github.com/pkg/errors"
+)
+
+const (
+	KubernetesConnectionSchemaID = "app:dynatrace.kubernetes.connector:connection"
+	KubernetesConnectionVersion  = "0.1.2"
+	KubernetesConnectionScope    = "environment"
+)
+
+type EnvironmentSetting struct {
+	ObjectId      *string                 `json:"objectId"`
+	SchemaId      string                  `json:"schemaId"`
+	SchemaVersion string                  `json:"schemaVersion"`
+	Scope         string                  `json:"scope"`
+	Value         EnvironmentSettingValue `json:"value"`
+}
+
+type EnvironmentSettingValue struct {
+	Name      string `json:"name"`
+	Uid       string `json:"uid"`
+	Namespace string `json:"namespace"`
+	Token     string `json:"token"`
+}
+
+type environmentSettingsResponse struct {
+	Items      []EnvironmentSetting `json:"items"`
+	TotalCount int                  `json:"totalCount"`
+	PageSize   int                  `json:"pageSize"`
+}
+
+func (c *client) GetConnectionSetting(uid string) (EnvironmentSetting, error) {
+	settingsObjectsUrl := c.getSettingsObjectsUrl()
+
+	req, err := http.NewRequestWithContext(c.ctx, http.MethodGet, settingsObjectsUrl, nil)
+	if err != nil {
+		return EnvironmentSetting{}, fmt.Errorf("error initializing http request: %w", err)
+	}
+
+	q := req.URL.Query()
+	q.Add("schemaIds", KubernetesConnectionSchemaID)
+	q.Add("scopes", KubernetesConnectionScope)
+
+	req.URL.RawQuery = q.Encode()
+
+	response, err := c.httpClient.Do(req)
+	defer utils.CloseBodyAfterRequest(response)
+
+	if err != nil {
+		return EnvironmentSetting{}, fmt.Errorf("error making post request to dynatrace api: %w", err)
+	}
+
+	responseData, err := c.getServerResponseData(response)
+	if err != nil {
+		return EnvironmentSetting{}, fmt.Errorf("error getting server response data: %w", err)
+	}
+
+	var resDataJson environmentSettingsResponse
+
+	err = json.Unmarshal(responseData, &resDataJson)
+	if err != nil {
+		return EnvironmentSetting{}, fmt.Errorf("error parsing response body: %w", err)
+	}
+
+	for _, item := range resDataJson.Items {
+		if item.Value.Uid == uid {
+			return item, nil
+		}
+	}
+
+	return EnvironmentSetting{}, nil
+}
+
+func (c *client) CreateConnectionSetting(es EnvironmentSetting) error {
+	log.Info("Creating connection setting", "EnvironmentSetting", es)
```

Comment:
private part of the token is printed to log
```
{"level":"info","ts":"2024-06-04T11:35:52.309Z","logger":"edgeconnect-client","msg":"Creating connection setting","EnvironmentSetting":{"objectId":null,"schemaId":"app:dynatrace.kubernetes.connector:connection","schemaVersion":"0.1.2","scope":"environment","value":{"name":"ao-ec-provisioner","uid":"273ec656-603d-46c8-b5f5-5c47a6903dff","namespace":"dynatrace","token":"dt0e01.2Xxxx.WPyyy"}}}
```

Reply:
True! Thank you.
Fixed [here](https://github.com/Dynatrace/dynatrace-operator/pull/3206/commits/6a4b8f2e72b859f69c0e43e5aecf0d9b0c615800)